crates/red_knot_module_resolver/src/resolver.rs

```diff
@@ -1,28 +1,30 @@
 use std::ops::Deref;
+use std::sync::Arc;
 
-use ruff_db::file_system::{FileSystem, FileSystemPath, FileSystemPathBuf};
-use ruff_db::vfs::{system_path_to_file, vfs_path_to_file, VfsFile, VfsPath};
+use ruff_db::file_system::FileSystemPathBuf;
+use ruff_db::vfs::{vfs_path_to_file, VfsFile, VfsPath};
 
-use crate::module::{Module, ModuleKind, ModuleName, ModuleSearchPath, ModuleSearchPathKind};
+use crate::db::Db;
+use crate::module::{Module, ModuleKind};
+use crate::module_name::ModuleName;
+use crate::path::ModuleResolutionPathBuf;
 use crate::resolver::internal::ModuleResolverSearchPaths;
-use crate::Db;
+use crate::supported_py_version::{set_target_py_version, SupportedPyVersion};
 
-const TYPESHED_STDLIB_DIRECTORY: &str = "stdlib";
-
-/// Configures the module search paths for the module resolver.
+/// Configures the module resolver settings.
 ///
 /// Must be called before calling any other module resolution functions.
 pub fn set_module_resolution_settings(db: &mut dyn Db, config: ModuleResolutionSettings) {
     // There's no concurrency issue here because we hold a `&mut dyn Db` reference. No other
     // thread can mutate the `Db` while we're in this call, so using `try_get` to test if
     // the settings have already been set is safe.
+    let (target_version, search_paths) = config.into_ordered_search_paths();
     if let Some(existing) = ModuleResolverSearchPaths::try_get(db) {
-        existing
-            .set_search_paths(db)
-            .to(config.into_ordered_search_paths());
+        existing.set_search_paths(db).to(search_paths);
     } else {
-        ModuleResolverSearchPaths::new(db, config.into_ordered_search_paths());
+        ModuleResolverSearchPaths::new(db, search_paths);
     }
+    set_target_py_version(db, target_version);
```

Comment:
We should try to avoid creating unnecessary salsa ingredients. Querying them, persisting them, and keeping them in memory all comes at a cost. I think I would aim for a single input ingredient for everything module resolver related (which kind of comes back that `ModuleResolutionSettings` maybe should be that input and it gets created with a builder)